src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenProjectConverter.java

```diff
@@ -196,13 +196,25 @@ private static MavenProject findMavenProject(final File modulePath, Collection<M
 
     File canonical = modulePath.getCanonicalFile();
     if (canonical.isDirectory()) {
-      canonical = new File(canonical, "pom.xml");
-    }
-    for (MavenProject module : modules) {
-      if (module.getFile().getCanonicalFile().equals(canonical)) {
-        return module;
+      File pom = new File(canonical, "pom.xml");
+      for (MavenProject module : modules) {
+        if (module.getFile().getCanonicalFile().equals(pom)) {
+          return module;
+        }
+      }
+      for (MavenProject module : modules) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Combine this loop with the one that starts on line 200. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3047)